crates/red_knot/src/lint.rs

```diff
@@ -86,7 +87,7 @@ pub(crate) fn lint_semantic(db: &dyn LintDb, file_id: FileId) -> QueryResult<Dia
         let context = SemanticLintContext {
             file_id: *file_id,
             source,
-            parsed,
+            parsed: &*parsed,
```

Comment:
Why do we need the reborrow here?

Reply:
`parsed` returns an `Arc<Parsed<...>>`. We could either pass in the owned Arc, or just the reference as I do here because we never need to clone (and the context already has lifetimes anyway)